common/src/main/java/com/revenuecat/purchases/common/OfferingFactories.kt

```diff
@@ -52,37 +57,17 @@ fun JSONObject.createOffering(productsById: Map<String, List<StoreProduct>>): Of
 
 fun JSONObject.createPackage(
     productsById: Map<String, List<StoreProduct>>,
-    offeringIdentifier: String
+    offeringIdentifier: String,
+    productMatchingMethod: (Map<String, List<StoreProduct>>, String, String?) -> StoreProduct?
 ): Package? {
     val packageIdentifier = getString("identifier")
     val productIdentifier = getString("platform_product_identifier")
-    val planIdentifier = optString("platform_product_plan_identifier").takeIf { it.isNotEmpty() }
 
-    val matchingProduct = getMatchingProduct(productsById, productIdentifier, planIdentifier)
-
-    return matchingProduct?.let { product ->
-        val packageType = packageIdentifier.toPackageType()
-        return Package(packageIdentifier, packageType, product, offeringIdentifier)
-    }
-}
+    val planIdentifier = optString("platform_product_plan_identifier").takeIf { it.isNotEmpty() }
+    val product = productMatchingMethod.invoke(productsById, productIdentifier, planIdentifier)
```

Comment:
if we went this route i'd need to think through thread-safety

Reply:
i don't think there are right now, but mayyyybe there could be in the future, so was thinking we should just be defensive? might refactor this anyway...